src/firefox/index.js

```diff
@@ -239,8 +244,32 @@ export type UseProfileParams = {
   app?: PreferencesAppName,
   configureThisProfile?: ConfigureProfileFn,
   customPrefs?: FirefoxPreferences,
+  createProfileFinder?: typeof defaultCreateProfileFinder,
 };
 
+export function defaultCreateProfileFinder(userDirectoryPath: string = '') {
+  const finder = new FirefoxProfile.Finder(userDirectoryPath);
+  const readProfiles = promisify(finder.readProfiles, finder);
+  return {
+    getPath: promisify(finder.getPath, finder),
+    hasProfileName: async (profileName: string) => {
+      try {
+        await fs.stat(path.join(FirefoxProfile.Finder.locateUserDirectory(),
```

Comment:
@harikishen the tests are failing because on an issue in the `hasProfileName` implementation:

- while we are configuring the Finder to use the specified parameter as the profiles search path (as expected)
  at line 251, here we are always doing an fs.stat on the default search path the `FirefoxProfile.Finder.locateUserDirectory()` provides (but it should use the same path that the `Finder` is going to use

And so it should be something like:
```js
        const profilesIniPath = path.join(
          userDirectoryPath || FirefoxProfile.Finder.locateUserDirectory(),
          'profiles.ini'
        );
        const res = await fs.stat(profilesIniPath);
```